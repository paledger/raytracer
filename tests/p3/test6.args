pixeltrace ../resources/refract2.pov 640 480 220 240
