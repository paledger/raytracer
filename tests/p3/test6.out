Pixel: [220, 240] Color: (83, 14, 84)
o - Iteration type: Primary
|   Ray: {0 0 14} -> {-0.203 0.00102 -0.9792}
|   Hit Object ID (1 - Sphere) at T = 12.75, Intersection = {-2.587 0.013 1.518}
|   Normal {-0.8624 0.004334 0.5061}
|   Transformed Ray: {0 0 14} -> {-0.203 0.00102 -0.9792}
|   Ambient: 0.1, 0, 0.1
|   Diffuse: 0.1179, 0, 0.1179
|   Specular: 0, 0, 0
|
|\
| o - Iteration type: Refraction
| |   Ray: {-2.587 0.013 1.517} -> {0.2449 -0.001231 -0.9695}
| |   Hit Object ID (1 - Sphere) at T = 4.211, Intersection = {-1.556 0.007817 -2.565}
| |   Normal {-0.5186 0.002606 -0.855}
| |   Transformed Ray: {-2.587 0.013 1.517} -> {0.2449 -0.001231 -0.9695}
| |   Ambient: 0.1, 0, 0.1
| |   Diffuse: 0, 0, 0
| |   Specular: 0, 0, 0
| |   Extra Info: into-object 
| |
| |\
| | o - Iteration type: Refraction
| | |   Ray: {-1.555 0.007814 -2.566} -> {0.6437 -0.003235 -0.7653}
| | |   Hit Object ID (2 - Sphere) at T = 2.357, Intersection = {-0.03813 0.0001916 -4.369}
| | |   Normal {-0.8861 -0.4347 -0.1606}
| | |   Transformed Ray: {-1.555 0.007814 -2.566} -> {0.6437 -0.003235 -0.7653}
| | |   Ambient: 0.152, 0.276, 0.16
| | |   Diffuse: 0.07356, 0.1336, 0.07743
| | |   Specular: 0, 0, 0
| | |   Extra Info: into-air 
| | |----
| | 
|  \
|   o - Iteration type: Reflection
|   |   Ray: {-1.556 0.007817 -2.565} -> {0.973 -0.004889 0.2309}
|   |   Hit Object ID (3 - Plane) at T = 819.7, Intersection = {796 -4 186.7}
|   |   Normal {0 1 0}
|   |   Transformed Ray: {-1.556 0.007817 -2.565} -> {0.973 -0.004889 0.2309}
|   |   Ambient: 0.08, 0.08, 0.32
|   |   Diffuse: 0.02754, 0.02754, 0.1102
|   |   Specular: 0, 0, 0
|   |----
|   
 \
  o - Iteration type: Reflection
  |   Ray: {-2.587 0.013 1.518} -> {-0.7559 0.003798 -0.6547}
  |   No intersection.
  |----
  

--------------------------------------------------------------------------------

